projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-site-migration.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Allow us to access the Migrate Guru site migration key via API.
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ */
+class WPCOM_REST_API_V2_Endpoint_Site_Migration {
+	/**
+	 * Class constructor
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'site-migration';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register our routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
```

Comment:
I think we need to make this API much more specific in terms of the path, as the `site-migration` path could encompass many more means and use cases, and this API is returning a super-specific piece of data.
I also think we should include a migration type parameter, as we may want to fetch different details/information depending on the migration plugin present on the site.

As a separate concern, I also believe that white labelled versions of the Migrate Guru plugin also use these classes, just with different configuration data, so I would also be wary of returning this information in cases where we don't think the Migrate Guru plugin is installed.